backend/icpy/api/rest_api.py

```
@@ -579,51 +653,81 @@ async def create_file(request: FileOperationRequest):
         async def delete_file(path: str):
             """Delete file."""
             try:
-                await self.filesystem_service.delete_file(path)
+                fs = self.filesystem_service
+                if self.context_router is not None:
+                    try:
+                        fs_rt = await self.context_router.get_filesystem()
+                        if getattr(fs_rt, 'is_remote', False):
+                            fs = fs_rt
+                    except Exception:
+                        pass
```

Ruff (0.13.2)

`try`-`except`-`pass` detected, consider logging the exception

(S110)